gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/CompileMetricSchemaTask.java

```diff
@@ -27,32 +26,34 @@
 import com.palantir.metric.schema.MetricSchema;
 import java.io.File;
 import java.io.IOException;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.ConfigurableFileCollection;
 import org.gradle.api.file.RegularFileProperty;
 import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.InputFiles;
 import org.gradle.api.tasks.OutputFile;
-import org.gradle.api.tasks.SourceTask;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
 import org.gradle.api.tasks.TaskAction;
 
 @CacheableTask
-public class CompileMetricSchemaTask extends SourceTask {
+public abstract class CompileMetricSchemaTask extends DefaultTask {
     private static final ObjectReader reader = ObjectMappers.withDefaultModules(new ObjectMapper(new YAMLFactory()))
             .readerFor(MetricSchema.class);
-    private static final ObjectWriter writer =
```

Comment:
why this?

Reply:
gotta save that object mapper initialization cost